internal/pkg/cli/svc_init.go

```diff
@@ -170,6 +170,8 @@ func newInitSvcOpts(vars initSvcVars) (*initSvcOpts, error) {
 		mftReader:    ws,
 		dockerEngine: dockerengine.New(exec.NewCmd()),
 	}
+	// Override the app flag since we don't only allow running this command within a workspace.
+	opts.appName = tryReadingAppName()
```

Comment:
do you think we should return an error if the flag value is different than the workspace app name? I wonder if it would be surprising that I specified `copilot svc init -a wrongapp` and copilot just changes the app name for me

Reply:
good idea!

Reply:
Ha, jinx! I suggested logging a msg, above, but agree that erroring out would be good.